Library/Homebrew/rubocops/cask/array_alphabetization.rb

```diff
@@ -0,0 +1,38 @@
+# typed: true
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Cask
+      class ArrayAlphabetization < Base
+        extend AutoCorrector
+
+        def on_send(node)
+          return if node.method_name != :zap
+
+          node.each_descendant(:pair).each do |pair|
+            pair.each_descendant(:array).each do |array|
+              if array.children.length == 1
+                add_offense(array, message: "Remove the `[]` around a single `zap trash` path") do |corrector|
+                  corrector.replace(array.source_range, array.children.first.source)
+                end
+              end
+
+              array.each_descendant(:str).each_cons(2) do |first, second|
+                next if first.source < second.source
+
+                add_offense(second, message: "The `zap trash` paths should be in alphabetical order") do |corrector|
+                  corrector.insert_before(first.source_range, second.source)
+                  corrector.insert_before(second.source_range, first.source)
+                  # Using `corrector.replace` here trips the clobbering detection.
+                  corrector.remove(first.source_range)
+                  corrector.remove(second.source_range)
+                end
+              end
```

Comment:
@issyl0 I could be over simplifying the process here, but in an effort to try to learn how RuboCop works better I thought I would try something here.

```suggestion
              next if array.children.length <= 1

              sorted_array = array.children.sort_by { |child| child.source.downcase }
              next if sorted_array.map(&:source) == array.children.map(&:source)

              add_offense(array,
                          message: "The `zap trash` paths should be in alphabetical order") do |corrector|
                array.children.each_with_index do |child, index|
                  corrector.replace(child.source_range, sorted_array[index].source)
                end
              end
```

Reply:
This is great! But it's a bit less specific about which lines are out of order - but that probably means fewer clobberings or weird results, so I'm inclined to go with this!

```
zap trash: [
           ^ The `zap trash` paths should be in alphabetical order
```

Reply:
This now runs perfectly, thanks again!